be/src/storage/rowset/segment.cpp

```diff
@@ -208,7 +214,8 @@ Status Segment::_open(size_t* footer_length_hint, const FooterPointerPB* partial
     SegmentFooterPB footer;
     RandomAccessFileOptions opts{.skip_fill_local_cache = skip_fill_local_cache};
     ASSIGN_OR_RETURN(auto read_file, _fs->new_random_access_file(opts, _fname));
-    RETURN_IF_ERROR(Segment::parse_segment_footer(read_file.get(), &footer, footer_length_hint, partial_rowset_footer));
+    RETURN_IF_ERROR(Segment::parse_segment_footer(read_file.get(), &footer, footer_length_hint, partial_rowset_footer,
```

Comment:
starlet will get file size internally for every `read(void *data, size_t count)`, you need to pass file_size to starlet when calling `_fs->new_random_access_file(opts, _fname)`